src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportOrchestratorTask.cs

```diff
@@ -301,6 +316,135 @@ private async Task UpdateProgressAsync(ImportOrchestratorTaskContext context, Ca
+                {
+                    TaskInfo latestTaskInfo = await _taskManager.GetTaskAsync(taskId, cancellationToken);
+
+                    if (latestTaskInfo.Status == TaskStatus.Completed)
+                    {
+                        CheckTaskResult(latestTaskInfo);
+                        completedTaskIds.Add(taskId);
+                    }
+                }
+
+                if (completedTaskIds.Count > 0)
+                {
+                    _orchestratorTaskContext.RunningTaskIds.RemoveAll(id => completedTaskIds.Contains(id));
+                    await UpdateProgressAsync(_orchestratorTaskContext, cancellationToken);
+                }
+
+                await Task.Delay(TimeSpan.FromSeconds(PollingFrequencyInSeconds), cancellationToken);
+            }
+        }
+
+        private async Task<(string taskId, long endSequenceId, long blobSizeInBytes)> CreateNewProcessingTaskAsync(Models.InputResource input, CancellationToken cancellationToken)
+        {
+            string processingTaskId = $"{_orchestratorInputData.TaskId}_{_orchestratorTaskContext.CreatedTaskCount}";
+
+            Dictionary<string, object> properties = await _integrationDataStoreClient.GetPropertiesAsync(input.Url, cancellationToken);
+            long blobSizeInBytes = (long)properties[IntegrationDataStoreClientConstants.BlobPropertyLength];
+            long estimatedResourceNumber = CalculateResourceNumberByResourceSize(blobSizeInBytes, DefaultResourceSizePerByte);
+            long beginSequenceId = _orchestratorTaskContext.CurrentSequenceId;
+            long endSequenceId = beginSequenceId + estimatedResourceNumber;
+
+            ImportProcessingTaskInputData importTaskPayload = new ImportProcessingTaskInputData()
+            {
+                ResourceLocation = input.Url.ToString(),
+                UriString = _orchestratorInputData.RequestUri.ToString(),
+                BaseUriString = _orchestratorInputData.BaseUri.ToString(),
+                ResourceType = input.Type,
+                TaskId = processingTaskId,
+                BeginSequenceId = beginSequenceId,
+                EndSequenceId = endSequenceId,
+            };
+
+            TaskInfo processingTask = new TaskInfo()
+            {
+                QueueId = _orchestratorInputData.ProcessingTaskQueueId,
+                TaskId = processingTaskId,
+                TaskTypeId = ImportProcessingTask.ImportProcessingTaskId,
+                InputData = JsonConvert.SerializeObject(importTaskPayload),
+                MaxRetryCount = _orchestratorInputData.ProcessingTaskMaxRetryCount,
+                ParentTaskId = _orchestratorInputData.TaskId,
+            };
+
+            TaskInfo taskInfoFromServer = await _taskManager.GetTaskAsync(processingTaskId, cancellationToken);
+            if (taskInfoFromServer == null)
+            {
+                taskInfoFromServer = await _taskManager.CreateTaskAsync(processingTask, false, cancellationToken);
+            }
```

Comment:
If the task was created earlier, do we need to update the task here?

Reply:
We still need to check it status, like running task.